aten/src/ATen/Exceptions.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include <ATen/ATen.h>
+
+#include <typeinfo>
+
+
+namespace at {
+namespace checks {
+
+inline bool tensor_on_same_device(const at::Tensor &ten1, const at::Tensor &ten2){
+  return ten1.get_device()==ten2.get_device();
+}
+
+inline bool tensor_on_same_device(
+  const at::Tensor &ten1,
+  const c10::optional<at::Tensor> &ten2
+){
+  return !ten2.has_value() || ten1.get_device()==ten2->get_device();
+}
+
+inline bool tensor_on_cpu(const at::Tensor &ten){
+  return !ten.is_cuda(); // TODO: Should be a better way to do this
```

Comment:
This is incorrect. For example, an XLA tensor is not on CUDA but is also not necessarily on the CPU.

Reply:
Acknowledged: I thought this might be an issue. It doesn't looks as though tensors have an `is_cpu` predicate, though.

Reply:
It looks like [this](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/templates/TensorMethods.cpp#L79) would be the place to add one, but it's unclear to me how to go about doing that.